src/main/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/controller/admin/PrisonExcludeDatesController.kt

```diff
@@ -0,0 +1,145 @@
+package uk.gov.justice.digital.hmpps.visitscheduler.controller.admin
+
+import io.swagger.v3.oas.annotations.Operation
+import io.swagger.v3.oas.annotations.media.Content
+import io.swagger.v3.oas.annotations.media.Schema
+import io.swagger.v3.oas.annotations.responses.ApiResponse
+import io.swagger.v3.oas.annotations.tags.Tag
+import jakarta.validation.Valid
+import org.springframework.http.MediaType
+import org.springframework.security.access.prepost.PreAuthorize
+import org.springframework.validation.annotation.Validated
+import org.springframework.web.bind.annotation.GetMapping
+import org.springframework.web.bind.annotation.PathVariable
+import org.springframework.web.bind.annotation.PutMapping
+import org.springframework.web.bind.annotation.RequestBody
+import org.springframework.web.bind.annotation.RequestMapping
+import org.springframework.web.bind.annotation.RestController
+import uk.gov.justice.digital.hmpps.visitscheduler.config.ErrorResponse
+import uk.gov.justice.digital.hmpps.visitscheduler.dto.PrisonDto
+import uk.gov.justice.digital.hmpps.visitscheduler.dto.PrisonExcludeDateDto
+import uk.gov.justice.digital.hmpps.visitscheduler.service.PrisonConfigService
+
+const val PRISONS_PATH: String = "/prisons"
+const val PRISON_EXCLUDE_DATE_PATH: String = "$PRISONS_PATH/prison/{prisonCode}/exclude-date"
+
+// TODO - remove STAFF from var name once we take ADMIN API endpoints to add / remove exclude dates out
+const val STAFF_ADD_PRISON_EXCLUDE_DATE: String = "$PRISON_EXCLUDE_DATE_PATH/add"
+
+// TODO - remove STAFF from var name once we take ADMIN API endpoints to add / remove exclude dates out
+const val STAFF_REMOVE_PRISON_EXCLUDE_DATE: String = "$PRISON_EXCLUDE_DATE_PATH/remove"
+const val GET_PRISON_EXCLUDE_DATES: String = PRISON_EXCLUDE_DATE_PATH
+
+@RestController
+@Validated
+@Tag(name = "Prison exclude dates rest controller")
+@RequestMapping(name = "Prison Exclude Dates Configuration Resource", produces = [MediaType.APPLICATION_JSON_VALUE])
+class PrisonExcludeDatesController(
```

Comment:
moved all exclude date endpoints into new controller and setting role as VISIT_SCHEDULER.